src/BaseActionWatcher.ts

```diff
@@ -48,7 +48,7 @@ export class BaseActionWatcher {
   protected async checkForBlocks(isReplay: boolean = false) {
     let headBlockNumber = 0
     while (!headBlockNumber || this.actionReader.currentBlockNumber < headBlockNumber) {
-      const [blockData, blockMeta] = await this.actionReader.nextBlock()
+      const [blockData, blockMeta, lastIrreversibleBlockNumber] = await this.actionReader.nextBlock()
```

Comment:
same in usage here, I think destructuring might be cleaner and returning an object in the former since array order may not always be guaranteed.

Reply:
This is destructuring- array destructuring. [Order is guaranteed because of typescript.](https://github.com/EOSIO/demux-js/blob/develop/src/AbstractActionReader.ts#L59)

Still, I'm not opposed to it. I guess the resistance comes from having to make a new interface for  it, which currently seems unnecessary. We're destructuring it immediately anyway, whether it's array or object destructuring. Maybe @jeffreyssmith2nd can weigh in his opinion.

Reply:
I think an object with an interface would be a little cleaner.